src/Std/Data/HashMap/Basic.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+prelude
+import Std.Data.DHashMap.Basic
+
+set_option linter.missingDocs true
+set_option autoImplicit false
+
+/-!
+# Hash maps
+
+This file develops the type `Std.Data.HashMap` of dependent hash maps.
+
+The operations `map` and `filterMap` on `Std.Data.HashMap` are defined in the module
+`Std.Data.HashMap.AdditionalOperations`.
+
+Lemmas about the operations on `Std.Data.HashMap` are available in the
+module `Std.Data.HashMap.Lemmas`.
+
+See the module `Std.Data.HashMap.Raw` for a variant of this type which is safe to use in
+nested inductive types.
+-/
+
+universe u v w
+
+variable {α : Type u} {β : Type v}
+
+namespace Std
+
+/-- Hash maps. -/
+structure HashMap (α : Type u) (β : Type v) [BEq α] [Hashable α] where
+  /-- Internal implementation detail of the hash map -/
+  inner : DHashMap α (fun _ => β)
```

Comment:
Should the constructor be made private?

Reply:
Unfortunately, we need to call this constructor in the file `HashMap/AdditionalOperations.lean`. We can make it private for `HashMap.Raw`, `HashSet` and `HashSet.Raw`, though.

Reply:
Actually, this is not true, none of the constructors can be private because `Lemmas.lean`/`RawLemmas.lean` needs to refer to them.